### PR TITLE
fix: treemap error(non boolean attribute z)

### PR DESCRIPTION
**Pull request by proke03**
## Description
That variable assignment gives the attribute z to the path element in the svg.
However, the expression is not valid because svg cannot set the z-value.
https://www.w3.org/TR/SVG2/render.html#RenderingOrder

## Related Issue
#2952

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/chart/Treemap.tsx [hunk 1]
```diff
@@ -108,8 +108,6 @@ const horizontalPosition = (row: any, parentSize: number, parentRect: TreemapNod
     child.width = Math.min(rowHeight ? Math.round(child.area / rowHeight) : 0, parentRect.x + parentRect.width - curX);
     curX += child.width;
   }
-  // what's z
```
Comment (nikolasrieble) [inline src/chart/Treemap.tsx:111]: Amazing comment. While this seems to be a bug fix, I am yet worried whether some other place in recharts magically expects this to be set. Please do manually test the TreeMap in the Demo, and investigate whether this is used anywhere.

Typescript does not help a lot due to the large amount of any's in the code base.

Comment (proke03) [reply src/chart/Treemap.tsx:111]: Thanks for your review.
Sure, I tested it in the Demo. I checked the result rendered and element attributes.

Comment (ckifer) [reply src/chart/Treemap.tsx:111]: +1 to Niko - I'm going to wait to merge this until I pull it down and play with it (at least just for some more confidence if nothing else). Looks like a non-breaking change from the naked eye (and testing). Thanks

**Comment by ckifer** [comment]
Pulled this down, it looks good to me

**Review by ckifer — APPROVED** [review]

**Comment by ckifer** [comment]
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/z 

The only valid elements for this we will never use

**Pull request merged**